sorting/radix_sort/radix_sort.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+    Radix Sort Algorithm
+    --------------------
+
+    Time Complexity - O(d*(n + b)),
+        where d is the max number of digits, n is the size of the input & b is the base of the system.
+
+    Space Complexity - O(n + b),
+        where n is the size of the input & b is the base of the system.
+*/
+
+#include <iostream>
+#include <vector>
+#include <cstdlib>
+#include <ctime>
+
+using namespace std;
+
+// Gets the maximum element in array (To find the max number of digits).
+int getMax(vector<int>& arr) {
+    if (arr.empty()) { 
+        cerr << "The Array to Sort is Empty!" << endl; 
+        exit(1);
+    }
+    int max = arr[0];
+    for (int num : arr) {
+        if (num > max) {
+            max = num;
+        }
+    }
+    return max;
+}
+
+// Count Sort Algorithm.
+void countSort(vector<int>& arr, int pos) {
+    int size = arr.size();
+    vector<int> digit_frequency(10, 0); // Sorting decimal numbers here, hence base is 10.
+
+    // Increments the freq of elements in each bucket.
+    for (int num : arr) {
+        ++digit_frequency[(num/pos) % 10]; 
+    }
+
+    // Calculates the cumulative frequency.
+    for (int i = 1; i < 10; i++) {
+        digit_frequency[i] += digit_frequency[i - 1];
+    }
+
+    // Builds the sorted array from the right to maintain the stability property of Radix Sort.
+    vector<int> sorted_arr(size);
+    for (int i = size-1; i >= 0; i--) {
+        int idx = (arr[i]/pos) % 10;
+        sorted_arr[--digit_frequency[idx]] = arr[i]; 
+    }
+
+    arr = sorted_arr; // Copies the sorted array to the original array.
+}
+
+// Radix Sort Algorithm.
+void radixSort(vector<int>& arr) {
+    int max = getMax(arr);
+    for(int pos = 1; max/pos > 0; pos *= 10) { 
+        countSort(arr, pos); // Calls countSort() for every digit position.
+    }
+}
+
+// Helper function to print elements of array.
+void printArray(int arr[], int size) {
```

Comment:
instead of passing in the entire array as object. When you use vector below, pass reference pointer here. memory saved!